### PR TITLE
[KernelGen][Nvidia] Add binary_cross_entropy operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `binary_cross_entropy`. Measures the binary cross entropy between the target and input probabilities, with optional per-element weighting.

## Testing
- Validated against PyTorch reference on device side
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_binary_cross_entropy.py --level core` (NVIDIA H20)

| Configuration | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [64, 64] fp16 | 0.012 | 0.013 | 0.955 |
| [256, 256] fp16 | 0.017 | 0.013 | 1.329 |
| [1024, 1024] fp16 | 0.022 | 0.017 | 1.313 |
| [4096, 4096] fp16 | 0.120 | 0.072 | 1.662 |
| [1024, 65536] fp16 | 0.427 | 0.245 | 1.742 |
| [64, 64] fp32 | 0.013 | 0.010 | 1.287 |
| [256, 256] fp32 | 0.019 | 0.011 | 1.731 |
| [1024, 1024] fp32 | 0.022 | 0.015 | 1.410 |
| [4096, 4096] fp32 | 0.116 | 0.074 | 1.565 |
| [1024, 65536] fp32 | 0.491 | 0.258 | 1.907 |
| [64, 64] bf16 | 0.012 | 0.013 | 0.899 |
| [256, 256] bf16 | 0.019 | 0.013 | 1.445 |
| [1024, 1024] bf16 | 0.023 | 0.017 | 1.311 |
| [4096, 4096] bf16 | 0.124 | 0.075 | 1.665 |
| [1024, 65536] bf16 | 0.441 | 0.252 | 1.748 |
| **Arithmetic Mean** | — | — | **1.443** |

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (30 cases, --level core) | 1.443 | Primary |
| Tianshu | FAIL | PASS | 1.988 | — |
| Muxi | FAIL | PASS | 1.215 | — |
| Ascend | FAIL | N/A | — | — |
| Hygon | FAIL | PASS | 0.622 | — |

## Files Changed
- `src/flag_gems/ops/binary_cross_entropy.py`: Triton kernel implementation
- `tests/test_binary_cross_entropy.py`: Accuracy test
- `benchmark/test_binary_cross_entropy.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

File: conf/operators.yaml [hunk 1]
```diff
@@ -822,6 +822,21 @@ ops:
       - Reduction
     stages:
       - stable: '5.0'
+  - id: binary_cross_entropy
```
Comment (tengqm) [inline conf/operators.yaml]: The protocol is to explicitly add a `binary_cross_entropy_out` after this one.

File: benchmark/test_binary_cross_entropy.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    bench = base.GenericBenchmark2DOnly(
+        op_name="binary_cross_entropy",
+        input_fn=binary_cross_entropy_input_fn,
+        torch_op=torch.nn.functional.binary_cross_entropy,
+        dtypes=consts.FLOAT_DTYPES,
+    )
+    bench.run()
+
+
+@pytest.mark.binary_cross_entropy
```
Comment (tengqm) [inline benchmark/test_binary_cross_entropy.py]:
```suggestion
@pytest.mark.binary_cross_entropy_out
```

File: benchmark/test_binary_cross_entropy.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+def test_binary_cross_entropy_out():
+    bench = base.GenericBenchmark2DOnly(
+        op_name="binary_cross_entropy",
+        input_fn=binary_cross_entropy_input_fn,
```
Comment (tengqm) [inline benchmark/test_binary_cross_entropy.py]: This input function will not provide the `out` keyword argument.

File: tests/test_binary_cross_entropy.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    ref_inp = utils.to_reference(inp, True)
+    ref_target = utils.to_reference(target, True)
+
+    ref_out = torch.nn.functional.binary_cross_entropy(
+        ref_inp, ref_target, reduction=reduction
+    )
+    with flag_gems.use_gems():
+        res_out = torch.nn.functional.binary_cross_entropy(
+            inp, target, reduction=reduction
+        )
```
Comment (tengqm) [inline tests/test_binary_cross_entropy.py:135]: Neither invocation provides the `out` parameter.

File: benchmark/test_binary_cross_entropy.py (hunk 2 above)
Comment (bwbwzzz) [reply benchmark/test_binary_cross_entropy.py]: thanks， have applied，

File: conf/operators.yaml (hunk 1 above)
Comment (bwbwzzz) [reply conf/operators.yaml]: I have added the corresponding description in the YAML file. Thank you for pointing this out.

File: benchmark/test_binary_cross_entropy.py (hunk 3 above)
Comment (bwbwzzz) [reply benchmark/test_binary_cross_entropy.py]: have reviesed，thanks

File: tests/test_binary_cross_entropy.py (hunk 4 above)
Comment (bwbwzzz) [reply tests/test_binary_cross_entropy.py:135]: thanks

**Review by bin913 — APPROVED** [review]
